testcontainers/src/core/client.rs

```diff
@@ -276,6 +290,55 @@ impl Client {
             .map_err(ClientError::StartContainer)
     }
 
+    pub(crate) async fn copy_to_container(
+        &self,
+        container_id: impl Into<String>,
+        target_path: &Path,
+        data: &[u8],
+    ) -> Result<(), ClientError> {
+        let container_id: String = container_id.into();
+
+        let parent = match target_path.parent() {
+            Some(v) => v,
+            None => {
+                return Err(ClientError::PathNameError(
+                    target_path.display().to_string(),
+                ))
+            }
+        };
+
+        let options = UploadToContainerOptions {
+            path: parent.display().to_string(),
+            no_overwrite_dir_non_dir: "false".into(),
+        };
+
+        let tar = Client::tar_data(data, target_path)?;
+
+        self.bollard
+            .upload_to_container::<String>(&container_id, Some(options), tar)
+            .await
+            .map_err(ClientError::CopyToContainer)
+    }
+
+    fn tar_data(data: &[u8], path: &Path) -> Result<Bytes, ClientError> {
+        let file_name = match path.file_name() {
+            Some(v) => v,
+            None => return Err(ClientError::PathNameError(path.display().to_string())),
+        };
+
+        let mut ar = tar::Builder::new(Vec::new());
+        let mut header = tar::Header::new_gnu();
+        header.set_path(file_name).map_err(ClientError::TarError)?;
+        header.set_size(data.len() as u64);
+        header.set_mode(0o0644);
+        header.set_cksum();
+        ar.append(&mut header, data)
+            .map_err(ClientError::TarError)?;
+        let bytes = ar.into_inner().map_err(ClientError::TarError)?;
+
+        Ok(Bytes::copy_from_slice(bytes.as_slice()))
+    }
```

Comment:
`tar` uses blocking api, we should avoid this by spawning blocking task at least.

Also it's not related to the `Client` itself

Reply:
I'll have a look at that.

Reply:
If you don't mind, I then just just this crate for taring the files: https://github.com/dignifiedquire/async-tar

Reply:
Yes, sure `async-tar` relies on `async-std`. We use tokio (at least for now, we don't provide a way to choose).

So I'd suggest to use https://crates.io/crates/tokio-tar (the same author, but for tokio)